src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -367,7 +369,20 @@ public List<DataFile> saveAndAddFilesToDataset(DatasetVersion version,
 						} else {
 							logger.fine("Failed to extract indexable metadata from file " + fileName);
 						}
-					} else if (FileUtil.MIME_TYPE_INGESTED_FILE.equals(dataFile.getContentType())) {
+                                        } else if (fileMetadataExtractableFromNetcdf(dataFile, tempLocationPath)) {
```

Comment:
This breaks direct uploads for me. When using a direct upload, the file is `unattached` and the owner is set to 'null' on line 336. However, 'fileMetadataExtractableFromNetcdf' needs the owner dataset to determine the folder inside `storageIO.open();`. This results in a `NullPointerException`, preventing adding the files to the dataset. I have only tested with the local filesystem, but from looking at the code, this probably also breaks uploading files directly to S3. Can you investigate and fix it? Thanks.

Reply:
@ErykKul yikes! This has already been merged. 😬  Could you please create an issue about this?

Reply:
I've reproduced this in develop: direct file upload via UI fails with a grayed out screen as if waiting for a completion signal and a server log error: 
[diredct_upload_ui_err.txt](https://github.com/IQSS/dataverse/files/11517681/diredct_upload_ui_err.txt)

Note that direct file upload continues to work via API. Also, deleting the file continues to work via ui.

Reply:
I can reproduce it too. I created an issue:

- #9601

Thanks, both!